dotnet/src/SemanticKernel/Orchestration/DefaultSKContext.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Semantic Kernel context.
+/// </summary>
+[DebuggerDisplay("{DebuggerDisplay,nq}")]
+internal sealed class DefaultSKContext : SKContext
+{
+    private CultureInfo _culture;
+    private bool _errorOccurred = false;
+    private Exception? _lastException;
+    private string _lastErrorDescription = string.Empty;
+    private IReadOnlyCollection<ModelResult> _modelResults;
+    private ContextVariables _variables;
+    private IReadOnlySkillCollection _skills;
+    private ILogger _logger;
+
+    internal static SKContext FromException(Exception exception)
+    {
+        DefaultSKContext context = new();
+        context.Fail(exception.Message, exception);
+        return context;
+    }
+
+    public override string Result => this.Variables.ToString();
+
+    public override bool ErrorOccurred => this._lastException != null;
+
+    public override string LastErrorDescription => this._lastErrorDescription;
```

Comment:
Even if we keep the error-related stuff on SKContext, do we need both LastErrorDescription and LastException?  If someone wants to provide a description, can they just provide an exception instead with the description as the message?

Reply:
No, we do absolutely do not. I had the same thought -- we should just remove this with the Exceptions work.